src/mscorlib/src/System/String.Searching.cs

```diff
@@ -96,8 +96,98 @@ public int IndexOfAny(char[] anyOf, int startIndex)
         }
 
         [Pure]
+        public int IndexOfAny(char[] anyOf, int startIndex, int count)
+        {
+            if (anyOf == null)
+            {
+                throw new ArgumentNullException(nameof(anyOf));
+            }
```

Comment:
It may not be common enough to warrant it (and thus may hurt the more common cases of 2/3), but I wonder if it's worthwhile adding here:
```C#
if (anyOf.Length == 1)
{
    return IndexOf(anyOf[0], startIndex, count);
}
```